api-gateway/src/main/resources/static/scripts/visits/visits.controller.js

```diff
@@ -267,11 +269,60 @@ angular.module('visits')
             }
         }
 
+        let sortStatusAscendingUpcomingVisits = false;
+        let sortStatusAscendingPreviousVisits = false;
+        self.SortTableByStatus = function(isForUpcoming) {
+            ResetSortButtonArrows(isForUpcoming);
+
+            if(isForUpcoming) {
+                sortStatusAscendingUpcomingVisits = !sortStatusAscendingUpcomingVisits;
+
+                if(sortStatusAscendingUpcomingVisits) {
+                    self.upcomingVisits.sort(function (a, b) {
+                        a = self.getStatus(a.status).toLowerCase();
+                        b = self.getStatus(b.status).toLowerCase();
+
+                        return a < b ? -1 : a > b ? 1 : 0;
+                    });
+                    $('#sortByStatusButtonUpcomingVisits').text("Sort by status ↓")
+                } else {
+                    self.upcomingVisits.sort(function (a, b) {
+                        a = self.getStatus(a.status).toLowerCase();
+                        b = self.getStatus(b.status).toLowerCase();
+
+                        return a > b ? -1 : a < b ? 1 : 0;
+                    });
+                    $('#sortByStatusButtonUpcomingVisits').text("Sort by status ↑")
+                }
+            } else {
+                sortStatusAscendingPreviousVisits = !sortStatusAscendingPreviousVisits;
+
+                if(sortStatusAscendingPreviousVisits) {
+                    self.previousVisits.sort(function (a, b) {
+                        a = self.getStatus(a.status).toLowerCase();
+                        b = self.getStatus(b.status).toLowerCase();
+
+                        return a < b ? -1 : a > b ? 1 : 0;
+                    });
+                    $('#sortByStatusButtonPreviousVisits').text("Sort by status ↓")
+                } else {
+                    self.previousVisits.sort(function (a, b) {
+                        a = self.getStatus(a.status).toLowerCase();
+                        b = self.getStatus(b.status).toLowerCase();
+
+                        return a > b ? -1 : a < b ? 1 : 0;
+                    });
+                    $('#sortByStatusButtonPreviousVisits').text("Sort by status ↑")
+                }
+            }
+        }
```

Comment:
There must be a way to call the same functions for sorting the previous visits table and scheduled visits table instead of calling a separate function every time.

Reply:
For both previous and scheduled, it will call the SortTableByStatus function, like Brandon mentioned, the boolean parameter decides whether it is previous or scheduled.